lib/clock.js

```diff
@@ -1,9 +1,19 @@
-function getLocalAdjustedTime(sdk) {
-  var localClockOffset = parseInt(sdk.options.localClockOffset || 0);
-  var now = (Date.now() + localClockOffset) / 1000;
-  return now;
+function SdkClock(localOffset) {
+  // Calculated local clock offset from server time (in milliseconds). Can be positive or negative.
+  this.localOffset = parseInt(localOffset || 0);
 }
 
-module.exports = {
-  getLocalAdjustedTime: getLocalAdjustedTime
+Object.assign(SdkClock.prototype, {
+  // Return the current time (in seconds)
+  now: function() {
+    var now = (Date.now() + this.localOffset) / 1000;
+    return now;
+  }
+});
+
+SdkClock.create = function(/* sdk, options */) {
+  // TODO: calculate localOffset
+  return new SdkClock();
```

Comment:
Doesn't this need to pass through the arguments sent to `create()`?

Reply:
it would pass the (calculated) local offset. Passing nothing means it is `0` Since it is internal/private, there is no way to set the localOffset value from the outside.

Reply:
Ah, so the idea is that we keep it "hidden" like this for now, maybe pass through in the future?

Reply:
yes, I've updated to show a zero value being passed explicitly. with a TODO explaining this